### PR TITLE
Minor fixes for upload component

### PR DESCRIPTION
**Pull request by HaoYuanTW**
## Context

I tested latest changes made here against send flow and discovered some problems
There is one console error i have addressed in the PR, also some subtle changes happened to angular digest. 

## Changes
Added asyncapply around some UI state changes
Fixed a miss routed parameter in callback

## Checklist

- [x] All changes are covered by tests

### CODE REVIEW

File: src/forms/form-control/form-control.controller.js [hunk 1]
```diff
@@ -36,9 +36,9 @@ class FormControlController {
   }
 
   // eslint-disable-next-line
-  onAsyncUploadSuccess(response) {
```
Comment (HaoYuanTW) [inline src/forms/form-control/form-control.controller.js]: response object here is a json containing response and also the file. therefore changing the wiring.

File: src/forms/upload/camera-capture/camera-capture.controller.js [hunk 2]
```diff
@@ -397,7 +395,9 @@ function createUploadCallback($ctrl) {
     }
     $ctrl.showVideoPreview = false;
     $ctrl.closeVideoStream();
-    $ctrl.onCapture({ file: blob });
+    $ctrl.$scope.$apply(() => {
+      $ctrl.onCapture({ file: blob });
```
Comment (HaoYuanTW) [inline src/forms/upload/camera-capture/camera-capture.controller.js]: without putting capture in an angular apply, this does not work in send flow. It works in demo app though.

File: src/forms/form-control/form-control.controller.js (hunk 1 above)
Comment (HaoYuanTW) [reply src/forms/form-control/form-control.controller.js]: this is causing a console error in send flow during testing by the way

Comment (StevePole) [reply src/forms/form-control/form-control.controller.js]: I disagree with the way of fixing this, we should pass multiple arguments, not an object.

File: src/forms/form-control/form-control.controller.js [hunk 3]
```diff
@@ -36,7 +36,7 @@ class FormControlController {
...
+  onAsyncUploadSuccess({ file, response }) {
```
Comment (HaoYuanTW) [inline src/forms/form-control/form-control.controller.js]: @StevePole like this?

**Pull request merged**